### PR TITLE
feat: group dashboard items by project

### LINKED ISSUE #233: Reorganize rally dashboard
Rally Dashboard

  Project           Issue/PR    Status              Changes   Age
❯ hyperlight-dev/hy PR #344     🟡 ready for review +84 -0    2h
  perlight-wasm

0 active · 1 done · 0 orphaned

Right now its a single table but I would like to be able to have grouped by project:

project 
  -> issue
  -> issue
Project2
  ->issue 
   -PR

### PR DESCRIPTION
**Pull request by jsturtevant**
## Summary

Reorganizes the Rally dashboard from a flat table to a grouped-by-project view.

### Changes
- **New groupByProject() utility** in dashboard-data.js — groups dispatches by repo name, preserving original order
- **DispatchTable component** — renders project name as bold yellow header rows with indented issue/PR rows underneath
- **Plain text  non-TTY output now shows grouped format with project headersdashboard** 
- **Keyboard navigation** preserved — up/down moves through all items across all groups seamlessly
- **Existing columns** (Issue/PR, Status, Changes, Age) maintained for each item row

### Tests
- 5 new groupByProject unit tests (grouping, ordering, edge cases)
- 3 new DispatchTable component tests (group headers, grouping behavior)
- Updated existing tests to match new layout

Closes #233

### CODE REVIEW

File: lib/ui/dashboard-data.js [hunk 1]
```diff
@@ -68,43 +83,37 @@ export function renderPlainDashboard({ project } = {}) {
...
-        (d.branch ?? '').padEnd(colWidths.branch),
-        truncatedFolder.padEnd(colWidths.folder),
-        (d.status ?? '').padEnd(colWidths.status),
-        (d.changes ?? '').padEnd(colWidths.changes),
-        age.padEnd(colWidths.age),
-      ].join(' ');
-      lines.push(row);
+    const groups = groupByProject(dispatches);
+    const colWidths = { issueRef: 12, branch: 28, folder: 30, status: 20, changes: 10, age: 6 };
+
```
[the remark this answers is not in the document]
Comment (jsturtevant) [reply lib/ui/dashboard-data.js]: Fixed — renderPlainDashboard() now uses groupByProject() with group headers and removed the Project column from individual rows.

File: lib/ui/dashboard-data.js [hunk 2]
```diff
@@ -68,43 +83,37 @@ export function renderPlainDashboard({ project } = {}) {
...
+          : folderPath;
+        const row = [
+          '  ',
+          issueRef.padEnd(colWidths.issueRef),
+          (d.branch ?? '').padEnd(colWidths.branch),
+          truncatedFolder.padEnd(colWidths.folder),
+          (d.status ?? '').padEnd(colWidths.status),
+          (d.changes ?? '').padEnd(colWidths.changes),
+          age.padEnd(colWidths.age),
+        ].join(' ');
```
[the remark this answers is not in the document]
Comment (jsturtevant) [reply lib/ui/dashboard-data.js]: Fixed — using template literals instead of join(' ') to avoid extra indentation spaces.

**Comment by jsturtevant** [comment]
Closing due to conflicts with recently merged PRs (#301, #304, #305). Will recreate from latest main.
